docs: add eoa login (#1696)
alchemyplatform/aa-sdk: docs/pages/react/login-methods/eoa-login.mdx

    enablePopupOauth: true, // must be set to "true" if you plan on using popup rather than redirect in the social login flow
    // optional config to override default session manager config
    sessionConfig: {
      expirationTimeMs: 1000 * 60 * 60, // 60 minutes (default is 15 min)
    },
  },
  {
    // authentication ui config - your customizations here
    auth: {
      sections: [
        [{ type: "email" }],
        [
          { type: "passkey" },
          { type: "social", authProviderId: "google", mode: "popup" },
          { type: "social", authProviderId: "facebook", mode: "popup" },
        ],
        [
          {
            type: "external_wallets",
            walletConnect: { projectId: "your-project-id" },
          },
        ],
      ],
      addPasskeyOnSignup: true,
      showSignInText: true,
    },
  },
);
```

In order to access other providers via `WalletConnect`, **also** add `walletConnect: { projectId: "your-project-id" }` to your config. You create a `WalletConnect` id [here](https://cloud.reown.com/sign-in).

```ts twoslash [src/config.ts]

> // @noErrors

Reviewer: whats this?

```tsx twoslash
// @noErrors
import { useConnect } from "@account-kit/react";

const { connectors, connect } = useConnect({
  onSuccess: (data) => {
    console.log("Connected!", data);
  },
  onError: (err) => {
    console.error("Connection failed", err);
  },
});

return (
  <div>
    {connectors.map((connector) => (
      <button key={connector.id} onClick={() => connect({ connector })}>
        Connect with {connector.name}
      </button>
    ))}
  </div>
);
```

### Bring in an EOA as a Smart Wallet Owner

For local wallets or JSON-RPC wallets that support the [EIP-1193](https://eips.ethereum.org/EIPS/eip-1193) standard (like MetaMask, Coinbase Wallet, etc.), you can use `WalletClientSigner` from `@aa-sdk/core` to bring in these EOAs as your smart wallet owner. More info [here](https://www.alchemy.com/docs/wallets/third-party/signers).
By making your EOA an owner of a smart account, you will have access to AA feature through your new smart wallet.

> ```ts

Reviewer: can we put twoslash on all code snippets?

> // @noErrors

Reviewer: This example should show actually connecting the signer to a smart account.

---
title: Connect with EOAs
description: How to connect to bring in your own EOAs to Account Kit
---

> You can log into Account Kit using your own EOA! This can be done using our pre-built UI components or custom UI.

Reviewer: ```suggestion
You can allow user's to login by connecting their existing EOA using Account Kit. This can be done using our pre-built UI components or custom UI - no need for a separate wallet connector! 
```

Depending on what method you choose, you can bring in your EOA:

> - As a "pure" EOA, meaning it will not have access to Account Abstraction (AA) features.

Reviewer: ```suggestion
- As a "pure" EOA, meaning users will use their EOA and the funds in that wallet as normal. Since it is an EOA and not a smart account, it will not have access to AA features like gas sponsorship or batching. 
```

> - As an owner and signer of your Alchemy smart account, which will give you access to AA features.

Reviewer: ```suggestion
- As an owner and signer of a new smart account. This will allow users to maintain self-custody and they will sign with their EOA for each transaction. They will not be able to use funds from the EOA directly, but their new wallet will get AA features like sponsorship and batching.
```

## Pre-built Components

Easily log in into your existing EOAs using popular wallet providers like **MetaMask**, **Rabby**, **Phantom**, and more.

> Note: You will not have access to Account Abstraction features here; this external, "pure" EOA will be purely used as an EOA.

Reviewer: ```suggestion

**Note:** Since this is not a smart account, users will not get sponsorship, batching, or other features, and instead they will use their EOA as normal. 
```

> Our SDK supports external wallet connection out of the box, so you can seamlessly support both Web2 users with social login and Web3 users who want to bring their own EOAs.We support enabling users to bring their existing EOA in 2 ways:

Reviewer: ```suggestion
Our SDK supports external wallet connection out of the box, so you can seamlessly support both Web2 users with social login and Web3 users who want to bring their own EOAs - no need for a different wallet connector. We support enabling users to bring their existing EOA in 2 ways:
```

1. **Automatically detect and display all wallet extensions available in the user's browser.** (_Note: if the user has no wallet extensions installed, nothing will be displayed_)
2. **WalletConnect for all other EOAs**

> With Account Kit, users can still connect to their existing wallets without sacrificing compatibility or convenience.

Reviewer: ```suggestion
```

Our hooks will handle checking if the connected wallet is a smart wallet or an existing EOA.

> Transactions can be signed and sent in the same way, but external EOA wallets don't have access to Account Abstraction benefits such as gas abstraction, batching, etc.

Reviewer: ```suggestion
```

> These EOAs will simply function as EOAs.

Reviewer: ```suggestion
```

> In order to access these external wallet providers, just make sure to include configure `type: "external_wallets"` in your config:

Reviewer: ```suggestion
In order to access these external wallet providers, include `type: "external_wallets"` in your config:
```

    storage: cookieStorage, // persist the account state using cookies (read more here: https://accountkit.alchemy.com/react/ssr#persisting-the-account-state)
    enablePopupOauth: true, // must be set to "true" if you plan on using popup rather than redirect in the social login flow
    // optional config to override default session manager config
    sessionConfig: {
      expirationTimeMs: 1000 * 60 * 60, // 60 minutes (default is 15 min)
    },
  },
  {
    // authentication ui config - your customizations here
    auth: {
      sections: [
        [{ type: "email" }],
        [
          { type: "passkey" },
          { type: "social", authProviderId: "google", mode: "popup" },
          { type: "social", authProviderId: "facebook", mode: "popup" },
        ],
        [
          {
            type: "external_wallets",
            walletConnect: { projectId: "your-project-id" },
          },
        ],
      ],
      addPasskeyOnSignup: true,
      showSignInText: true,
    },
  },
);
```

> In order to access other providers via `WalletConnect`, **also** add `walletConnect: { projectId: "your-project-id" }` to your config:

Reviewer: Add a link for how they can get their wallet connect id

```ts twoslash [src/config.ts]
// @noErrors
[
  {
    type: "external_wallets",
    walletConnect: { projectId: "your-project-id" },
  },
];
```

## Custom UI

### React Hooks

> If you need complete control over the user experience, you can customize your EOA login. This is a "pure" EOA and will not have access to AA features:

Reviewer: ```suggestion
If you need complete control over the user experience, you can customize your EOA connection.
```

> For example, using [useConnect](https://www.alchemy.com/docs/wallets/reference/account-kit/react/hooks/useConnect)

Reviewer: ```suggestion
For example, use the [useConnect](https://www.alchemy.com/docs/wallets/reference/account-kit/react/hooks/useConnect) hook to allow users to connect their EOA and use it within your dapp:
```

```tsx twoslash
// @noErrors
import { useConnect } from "@account-kit/react";

const { connectors, connect } = useConnect({
  onSuccess: (data) => {
    console.log("Connected!", data);
  },
  onError: (err) => {
    console.error("Connection failed", err);
  },
});

return (
  <div>
    {connectors.map((connector) => (
      <button key={connector.id} onClick={() => connect({ connector })}>
        Connect with {connector.name}
      </button>
    ))}
  </div>
);
```

> ### EIP-1193-Compatible Provider Integration

Reviewer: This is sort of confusing. We talk about the split between bring EOA as eoa and bring eoa as owner above. lets make the headers and options here align with that mental model - they dont need to know about eips top level

Reviewer: Something like, to allow users to connect their EOA and use it as an owner on a new smart account to get gas sponsorship etc. do xyz

For wallets that support the [EIP-1193](https://eips.ethereum.org/EIPS/eip-1193) standard (like MetaMask, Coinbase Wallet, etc.), you can use `WalletClientSigner` from `@aa-sdk/core`. More info [here](https://www.alchemy.com/docs/wallets/third-party/signers).
This EOA is made an owner of your smart account and has access to AA features:

```ts
// @noErrors
import { WalletClientSigner, type SmartAccountSigner } from "@aa-sdk/core";
import { createWalletClient, custom } from "viem";
import { sepolia } from "viem/chains";

const externalProvider = window.ethereum; // or another EIP-1193 provider

const walletClient = createWalletClient({
  chain: sepolia,
  transport: custom(externalProvider),
});

export const signer: SmartAccountSigner = new WalletClientSigner(
  walletClient,
  "json-rpc",
);
```

> ### Local Account Integration

Reviewer: Dont think we need this section

Reviewer: Id remove
